app/src/main/java/org/oppia/android/app/onboarding/CreateProfileFragmentPresenter.kt

```diff
@@ -0,0 +1,107 @@
+import org.oppia.android.util.parser.image.ImageViewTarget
+import javax.inject.Inject
+
+private const val GALLERY_INTENT_RESULT_CODE = 1
+
+/** Presenter for [CreateProfileFragment]. */
+@FragmentScope
+class CreateProfileFragmentPresenter @Inject constructor(
+  private val fragment: Fragment,
+  private val activity: AppCompatActivity,
+  private val createProfileViewModel: CreateProfileViewModel,
+  private val imageLoader: ImageLoader
+) {
+  private lateinit var binding: CreateProfileFragmentBinding
+  private lateinit var uploadImageView: ImageView
+  private lateinit var selectedImage: String
+
+  /** Initialize layout bindings. */
+  fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View {
+    binding = CreateProfileFragmentBinding.inflate(
+      inflater,
+      container,
+      /* attachToRoot= */ false
+    )
+    binding.let {
+      it.lifecycleOwner = fragment
+      it.viewModel = createProfileViewModel
+    }
+
+    uploadImageView = binding.createProfileUserImageView
+
+    uploadImageView.apply {
+      setColorFilter(
+        ResourcesCompat.getColor(
+          activity.resources,
+          R.color.component_color_avatar_background_25_color,
+          null
+        ),
+        PorterDuff.Mode.DST_OVER
+      )
+
+      imageLoader.loadDrawable(
+        R.drawable.ic_default_avatar,
+        ImageViewTarget(this)
+      )
+    }
+
+    binding.onboardingNavigationContinue.setOnClickListener {
+      val nickname = binding.createProfileNicknameEdittext.text.toString().trim()
+
+      createProfileViewModel.hasErrorMessage.set(nickname.isBlank())
+    }
+
+    binding.createProfileNicknameEdittext.addTextChangedListener(object : TextWatcher {
+      override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+      override fun afterTextChanged(s: Editable?) {}
+      override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+        createProfileViewModel.hasErrorMessage.set(false)
+      }
+    })
+
+    binding.onboardingNavigationBack.setOnClickListener { activity.finish() }
+    binding.createProfileEditPictureIcon.setOnClickListener { openGalleryIntent() }
+    binding.createProfilePicturePrompt.setOnClickListener { openGalleryIntent() }
+    binding.createProfileUserImageView.setOnClickListener { openGalleryIntent() }
+
+    return binding.root
+  }
+
+  /** Receive the result of image upload and load it into the image view. */
+  fun handleOnActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    if (requestCode == GALLERY_INTENT_RESULT_CODE && resultCode == Activity.RESULT_OK) {
+      binding.createProfilePicturePrompt.visibility = View.GONE
+      data?.let {
+        selectedImage = checkNotNull(data.data.toString()) { "Could not find the selected image." }
+        imageLoader.loadBitmap(
+          selectedImage,
+          ImageViewTarget(uploadImageView)
+        )
+      }
+    }
+  }
```

Comment:
I have two suggestions here:
1. Rename `data: Intent?` parameter to `intent: Intent?` to enhance code readability.
2. The code can be simplified a bit as shown below. We don't need to check for `data?.let {}`, since it will be checked automatically by the `checkNotNull` block.

```suggestion
  fun handleOnActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
    if (requestCode == GALLERY_INTENT_RESULT_CODE && resultCode == Activity.RESULT_OK) {
      binding.createProfilePicturePrompt.visibility = View.GONE
      selectedImage = checkNotNull(intent?.data.toString()) { "Could not find the selected image." }
      imageLoader.loadBitmap(
        selectedImage,
        ImageViewTarget(uploadImageView)
      )
    }
  }
```

Reply:
Done, thanks.